docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
-# Manifest factorization through `.flux.yaml` configuration files
+# Manifest generation through `.flux.yaml` configuration files
 
-## Enabling search of `.flux.yaml` files
+This feature lets you generate Kubernetes manifests with a program,
+instead of having to include them in your git repo as YAML files. For
+example, you can use `kustomize` to patch a common set of resources to
+suit a particular environment.
 
-To enable it supply the command-line flag `--manifest-generation=true` to `fluxd`.
+## Setting manifest generation up
 
-## Goal
+Manifest generation is controlled by the flags given to `fluxd`, and
+`.flux.yaml` files in your git repo.
 
-It is a common pattern to run very similar resources in separate clusters. There are various scenarios in which this is
-required, the two main ones being:
+To enable it, pass the command-line flag `--manifest-generation=true`
+to `fluxd`.
 
-* Having a staging/canary cluster and production cluster. The resources from the staging cluster are regularly promoted
-  to the production cluster. In addition, there are long-term differences between the cluster resources (e.g. different
-  security keys, different database endpoints etc ...).
-* Federation. Different clusters run in separate regions, usually with very similar resources but different
-  configurations.
+For each path supplied with the flag `--git-path` (or `.` if none are
+supplied),
 
-The main goal of `.flux.yaml` configuration files is to help deploying similar resources/clusters:
+ - if a `.flux.yaml` file is found, it is used **instead** of looking
+   for YAML files;
 
-* with minimal replication of resource definitions
-* while keeping Flux neutral about the factorization technology used
+ - if no `.flux.yaml` file is found, the usual behaviour of looking
+   for YAML files is used.
 
-## File-access behaviour in Flux
+The manifests from all the `--git-path` values -- read from YAML files
+or generated -- are combined before applying to the cluster. If
+duplicates are detected, an error is logged and the sync is abandoned.
```

Comment:
```suggestion
duplicates are detected, an error is logged and Flux will abort the synchronization with the cluster.
```

Reply:
Rephrased, though not exactly as suggested